Source/NSTextView.m

```diff
@@ -4982,6 +4982,44 @@ the range of characters (if any) to be replaced by the new data.  */
 	  RELEASE(attachment);
 	  return YES;
 	}
+      if ([type isEqualToString: NSFilenamesPboardType])
+	{
+          NSArray *list = [pboard propertyListForType: NSFilenamesPboardType];
+          NSMutableAttributedString *as = [[NSMutableAttributedString alloc] init]; 
+
+	  id<NSFastEnumeration> enumerator = list;
+	  FOR_IN (NSString*, filename, enumerator)
+	   {
+	      NSFileWrapper *fw = [[NSFileWrapper alloc] initWithPath: filename];
+	      if (fw) 
+	        {
+	          NSTextAttachment *attachment = [[NSTextAttachment alloc] 
+					         initWithFileWrapper: fw];
+	          NSAttributedString *asat =
+	            [NSAttributedString attributedStringWithAttachment: attachment];
+
+	          RELEASE(fw);
+	          RELEASE(attachment);
+
+	          [as appendAttributedString: asat];
+	        }
+	   }
+	  END_FOR_IN(enumerator)
+
+	  if (as && changeRange.location != NSNotFound &&
```

Comment:
What I still don't understand is what the check for ´as` is for? Do you want to check whether the value is null (which the code does) or are you expecting this to check if we did append anything the the above loop (which it doesn't)?

Reply:
'as' will get appended to/replace the main text (lines 5013+)

Reply:
I can see what you do with `as` later on in the code. The question was what this is about. Do you mean `as != nil` or `[as length] != 0` ?

Reply:
Ah, I see what you mean. It should be [as length]. My mistake.

Reply:
Could you please correct this? I really would like to see this merged.

Reply:
As nothing was changed here for some time I will go ahead and merge this and fix it on main.